app/src/pages/inside/projectSettingsPageContainer/content/notifications/utils.js

```diff
@@ -17,6 +17,15 @@
 import { OWNER } from 'common/constants/permissions';
 import { EMAIL } from 'common/constants/pluginNames';
 
+function getUpdatedProperties(base, updated) {
+  return Object.keys(base).reduce((acc, key) => {
+    if (updated[key] && base[key] !== updated[key]) {
+      acc[key] = updated[key];
+    }
+    return acc;
+  }, base);
+}
+
```

Comment:
Looks like we've come to this complication because of the:
1. We didn't specify the `ruleDetails` fields under the `ruleDetails` object within the form.

We can use
```javascript
<FieldElement
    name={`${RULE_DETAILS_FIELD_KEY}.${field.name}`}
    // ...
/>
```
where `RULE_DETAILS_FIELD_KEY = 'ruleDetails'`.
Instead of
```javascript
<FieldElement
    name={field.name}
    // ...
/>
```
in the `modals/addEditNotificationModal/addEditNotificationModal.jsx, line 372`. In this case the dynamic fields will be stored within `ruleDetails` object. Don't forget to adjust validation if needed.

2. We've added `ruleFields` to each rule as its own field that caused to form initialization with this value.

![image](https://github.com/reportportal/service-ui/assets/31861805/a2fb255e-f3af-4688-9a76-da60c413c403)

Here I suggest to add a new prop `ruleItemContentProps` or `ruleContentProps` to the `RuleList` component to destructure its value while rendering `ruleItemContent`.
In this case we can put `ruleFields` there and it will be available within `NotificationRuleContent` and won't be initialized within the form.

![image](https://github.com/reportportal/service-ui/assets/31861805/64c6b445-641d-4515-8350-afdae7e39ed9)

I suppose after that changes additional processing within submission preparation won't be required.